compiler/rustc_codegen_ssa/src/mir/block.rs

```diff
@@ -908,9 +917,49 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
             (args, None)
         };
 
+        let const_vec_arg_indexes = (|| {
+            if let Some(def) = def
+                && let Some(attr) =
+                    bx.tcx().get_attr(def.def_id(), sym::rustc_intrinsic_const_vector_arg)
+            {
+                attr.meta_item_list()
+                    .iter()
+                    .flatten()
+                    .map(|item: &NestedMetaItem| match item {
+                        NestedMetaItem::Lit(MetaItemLit {
+                            kind: LitKind::Int(index, _), ..
+                        }) => index.get() as usize,
+                        _ => span_bug!(item.span(), "attribute argument must be an integer"),
```

Comment:
create a helper for this and use it in both attr parsing sites

Reply:
While you're here can you explain to me why `bx.tcx().get_attr` on line 923 causes the failure I'm seeing in CI?

Reply:
`get_attr` within the same crate will create a dependency edge from the calling query to the HIR. It's not a query. Since this is only allowed on intrinsics, we could (after https://github.com/rust-lang/rust/pull/120675) store it in the `ty::IntrinsicDef` struct, so it gets encoded in metadata (so is checked across crates, too), and so it doesn't cause this undesirable dep graph edge

Reply:
This attribute is intended for architecture specific intrinsics as used [here](https://github.com/rust-lang/stdarch/pull/1503/files) rather than rust-intrinsics like `simd_shuffle` so I don't think that could be used.

Reply:
These should all at some point go through the `intrinsic` query, they just don't do it yet.

Reply:
Okay, is there an issue/PR for this that I can mark this PR as blocked on?

Reply:
https://github.com/rust-lang/rust/pull/120675 is what needs to land first

Reply:
So that means that all of the intrinsic functions in files like [this](https://github.com/rust-lang/stdarch/blob/master/crates/core_arch/src/arm_shared/neon/generated.rs) will need to be marked with `#[rustc_intrinsic]`? Is there already a PR for this or is that work that will need to be done to enable features such as those provided in this PR?

Reply:
Alternatively you can encode this information in `codegen_fn_attrs`, I think that should also avoid the dep graph edges

Reply:
I went with the option of encoding the information on `codegen_fn_attrs` and calling the function to retrieve the indices: 25f9653a2eb27f3d2eee5aa003f99087eed28670

I tried putting it on the `ty::IntrinsicDef` however it failed to compile with the below error message, perhaps I was doing something wrong. I settled for `codegen_fn_attrs` as it seemed like the path of least resistance:

```
error[E0204]: the trait `std::marker::Copy` cannot be implemented for this type
   --> compiler/rustc_middle/src/ty/intrinsic.rs:6:10
    |
6   | #[derive(Copy, Clone, Debug, Decodable, Encodable, HashStable)]
    |          ^^^^ in this derive macro expansion
...
13  |     pub const_vector_indices: Option<Vec<usize>>,
    |     -------------------------------------------- this field does not implement `std::marker::Copy`
```